hack/lib/start.sh

```diff
@@ -0,0 +1,496 @@
+	ps -ef | grep openshift
+
+	if [[ -z "${api_server_version}" && -z "${controllers_version}" && -z "${skip_node}" ]]; then
+		os::start::internal::start_all_in_one
+		return
+	fi
+
+	os::start::internal::start_etcd
+	os::start::internal::start_api_server "${api_server_version}"
+	os::start::internal::start_controllers "${controllers_version}"
+	if [[ -z "${SKIP_NODE}" ]]; then
+		os::start::internal::start_node
+	fi
+}
+readonly -f os::start::server
+
+# os::start::master starts the OpenShift master, exports the PID of the OpenShift master and waits until OpenShift master endpoints are available
+# It is advised to use this function after a successful run of 'os::start::configure_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+# Arguments:
+#  None
+# Returns:
+#  - export OS_PID
+function os::start::master() {
+	echo "[INFO] `openshift version`"
+	echo "[INFO] Server logs will be at:	${LOG_DIR}/openshift.log"
+	echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+	echo "[INFO] Config dir is:		${SERVER_CONFIG_DIR}"
+	echo "[INFO] Using images:		${USE_IMAGES}"
+	echo "[INFO] MasterIP is:		${MASTER_ADDR}"
+
+	mkdir -p ${LOG_DIR}
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
+	echo "[INFO] Starting OpenShift server"
+	local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
+	$(os::start::internal::openshift_executable) start master \
+		--config=${MASTER_CONFIG_DIR}/master-config.yaml \
+		--loglevel=4 --logspec='*importer=5' \
+	&>"${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
+
+	echo "[INFO] OpenShift server start at: "
+	date
+
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 160
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 160
+
+	echo "[INFO] OpenShift server health checks done at: "
+	date
+}
+readonly -f os::start::master
+
+# os::start::internal::start_all_in_one starts the OpenShift server all in one.
+# It is advised to use this function after a successful run of 'configure_os_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - VOLUME_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - NODE_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Returns:
+#  - export OS_PID
+function os::start::internal::start_all_in_one () {
+	local use_latest_images
+	if [[ -n "${USE_LATEST_IMAGES:-}" ]]; then
+		use_latest_images="true"
+	else
+		use_latest_images="false"
+	fi
+
+	echo "[INFO] Starting OpenShift server"
+	local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
+	$(os::start::internal::openshift_executable) start \
```

Comment:
Nit, but we could format these to be parallel to the others:
```sh
local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
local openshift_executable
openshift_executable="$(os::start::internal::openshift_executable)"
${openshift_executable} start                                                     \
                        --loglevel=4                                              \
                        --logspec='*importer=5'                                   \
                        --latest-images="${use_latest_images}"                    \
                        --node-config="${NODE_CONFIG_DIR}/node-config.yaml"       \
                        --master-config="${MASTER_CONFIG_DIR}/master-config.yaml" \
                        >"${LOG_DIR}/openshift.log" 2>&1 &
```

(although I like how you've inlined the `openshift_executable` call here, it would make the indent on the following lines super deep)